plugin/completion/base_complete.py

```diff
@@ -120,16 +120,20 @@ def init(self, view, settings):
 
         """
         if not view:
+            log.critical(" no view to initialize completer!")
             return
         current_dir = path.dirname(view.file_name())
         search_scope = SearchScope(
             from_folder=current_dir,
             to_folder=settings.project_base_folder)
         self.flags_manager = FlagsManager(
-            use_cmake=settings.generate_flags_with_cmake,
-            flags_update_strategy=settings.cmake_flags_priority,
-            cmake_prefix_paths=settings.cmake_prefix_paths,
+            view=view,
+            settings=settings,
+            compiler_variant=self.compiler_variant,
```

Comment:
Ok, the name sounds good. Better than `init`. Changing.

As for recreation of FlagsManager - I don't see a big problem here. Maybe it could be done more efficiently, but it is out of the scope of this pull request. We could open a new issue for this. As I feel the change will not be trivial.